src/handler.py

```diff
@@ -35,37 +32,43 @@
 
 
 try:
-    connection = 'mysql+pymysql://{user}:{password}@{host}/{db_name}'.format(user=rds_config.db_username, password=rds_config.db_password, host=rds_config.rds_host, db_name=rds_config.db_name)
+    connection = 'mysql+pymysql://{user}:{password}@{host}/{db_name}'.format(user=rds_config.db_username,
+                                                                             password=rds_config.db_password,
+                                                                             host=rds_config.rds_host,
+                                                                             db_name=rds_config.db_name)
     engine = create_engine(connection)
     alchemy_utils.Base.metadata.create_all(engine)
     Session = sessionmaker(bind=engine)
+    logger.info("SUCCESS: Connection to RDS mysql instance succeeded")
 except Exception as e:
     logger.error("ERROR: Unexpected error: Could not connect to MySql instance.\t{}".format(e))
     sys.exit()
 
 
-logger.info("SUCCESS: Connection to RDS mysql instance succeeded")
 def scrape(n_tries, departure_airport, arrival_airport, departure_date):
     for i in range(n_tries):
         try:
             legs = get_legs(departure_airport, arrival_airport, departure_date)
-            if legs == []:
-                raise ValueError("No results")
-            persist_legs(legs, departure_date)
-            return 'Scraping successful'
-        except Exception as e:
+            if legs is []:
+                raise ValueError("No results.")
+            persist_legs(legs)
+            return 'Scrape successful.'
+        except Exception as e:  # TODO: import the TimeoutException class and except it and ValueError explicitly
             logger.error('Attempt {} of {} failed. Retrying...'.format(i+1, n_tries))
             traceback.print_exc()
-    return logger.error('All attempts failed for DEP:\t{}\nARR:\t{}\nDate:\t{}'.format(departure_airport, arrival_airport, departure_date))
+    return logger.error('All attempts failed for DEP:\t{}\nARR:\t{}\nDate:\t{}'.format(departure_airport,
+                                                                                       arrival_airport,
+                                                                                       departure_date)
+                        )
 
 
-def persist_legs(legs, departure_date):
+def persist_legs(legs):
     session = Session()
     try:
         session.add_all(legs)
         session.commit()
         logger.info('Committed')
-    except:
+    except Exception:  # TODO: what exception is this trying to catch?
```

Comment:
Any- it should be left as except- if anything whatsoever bubbles up while we're trying to commit we should rollback the transaction

Reply:
Ok. Minor Python point: 'Exception' is any.

Reply:
Yes I know. Except: is the same as Except Exception:

Reply:
Also general point. The point of handling specific exceptions is so that you don't actually squash genuine errors, eg caused by a programming error. But here we reraise the exception anyway, we don't squash it. So there's no point in being specific.

Reply:
Why do we re-raise? Why not just allow it to raise the exception here?

Reply:
What? You're already handling it? You're in the except and you need to rollback the transaction before you reraise...

Reply:
You're right it should be except Exception: thought- turns out there's a load of really weird exceptions that don't sub-class Exception